plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -40,8 +41,9 @@ var _ = admission.Interface(&persistentVolumeLabel{})
 type persistentVolumeLabel struct {
 	*admission.Handler
 
-	mutex      sync.Mutex
-	ebsVolumes aws.Volumes
+	mutex            sync.Mutex
+	ebsVolumes       aws.Volumes
+	gceCloudProvider *gce.GCECloud
```

Comment:
If volumes are plugins and Kubelet (and others) don't know what plugins exist, anything that escapes those packages is leaky.

Would we instead want to give the same list of plugins to the API server that we give to Kubelet and ControllerManager?

The API server would do the same iteration and CanSupport on each plugin, find the appropriate one, and call Validate(pv) and Admit(pv).

Each volume would know how to validate itself in the API.  Missing plugins would cause a real API error.  Any business logic regarding labelling and so forth is also internalized to the plugin.

@thockin thoughts?